File: tam_bolenleri.go
package main

import (
	"fmt"
)

func tam_bolenler(sayi int) [] int{
	var sonuc []int

	for index := 1; index <= sayi; index++{
		if  sayi % index == 0{
			sonuc = append(sonuc, index, (-1)*index)
		}
	}
	return sonuc
}

func main(){
	fmt.Print("Lütfen tam bölenlerini görmek listediğiniz sayıyı giriniz:")
	var sayi int
	_, err := fmt.Scanf("%d", &sayi)
	if err != nil{
		fmt.Println("sayi giriniz..")
	}
	sonuc := tam_bolenler(sayi)
	fmt.Println(sonuc)
}

File: tam_bolenleri.py
def tam_bolenler(sayi):
    #index = (-1)*sayi
    index = 1
    sonuc = []
    while index <= sayi:
        if sayi % index == 0:
            sonuc.extend([index, (-1)*index])
        index += 1  

    return sonuc

sayi    = int(input("Lütfen tam bölenlerini görmek listediğiniz sayıyı giriniz:"))
sonuc   = tam_bolenler(sayi)
print(sonuc)